src/WebJobs.Script/Description/FunctionDescriptorProvider.cs

```diff
@@ -143,7 +144,19 @@ private bool TryParseTriggerParameter(JObject metadata, out ParameterDescriptor
                     attributeBuilders.Add(attributeBuilder);
                 }
 
-                Type triggerParameterType = parameterType ?? binding.DefaultType;
+                Type triggerParameterType = parameterType;
+                if (triggerParameterType == null)
+                {
+                    if (bindingContext.DataType == "HttpResponseMessage")
```

Comment:
Code like this doesn't belong here. It would go in the HttpScriptBinding.DefaultType property getter.